### PR TITLE
Refracto: update more code to ts

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Refracto to have more file in ts:
it's allow to have type-check and it's "securise" our code

> [!NOTE]
> There are still files written in javascript, either they are used in the `next.config.js` so they remain in js for the form but if SWC transpiles the ts. 
> And there are also node scripts, but you'd need to have a loader or be in node 22.
> So for the moment I'd rather do nothing for those.

## Validation

- All test should pass
- Website should display correctly (use Vercel preview)

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- **NA** I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Review by ovflowd — COMMENTED** [review]
FYI many of these files **are intentionally done in plain JS** -- so that they do not need to be transpiled by TypeScript.

There is no virtual benefit (necessarily) for having them in TypeScript. What's the motivation behind this change?

**Comment by AugustinMauroy** [comment]
> FYI many of these files **are intentionally done in plain JS** -- so that they do not need to be transpiled by TypeScript.
> 
> There is no virtual benefit (necessarily) for having them in TypeScript. What's the motivation behind this change?

For me, it's ‘important’ to have them in typescript, for example `next.dynamic`, which is responsible for the entire content of the site. Because typescript offers type-checking which allows us to check that our code is correct and is supposed to help with the review.
We shouldn't all write in typescript in ‘wow that's typescript’ mode but see it more as a tool to limit errors.

**Review by avivkeller — APPROVED** [review]
LGTM code wise, but I'd prefer to get other reviews before this is merged.

IMO there's not much difference between having these files in JS or TS. But, I'm curious to see some other opinions

**Comment by ovflowd** [comment]
> > FYI many of these files **are intentionally done in plain JS** -- so that they do not need to be transpiled by TypeScript.
> > There is no virtual benefit (necessarily) for having them in TypeScript. What's the motivation behind this change?
> 
> For me, it's ‘important’ to have them in typescript, for example `next.dynamic`, which is responsible for the entire content of the site. Because typescript offers type-checking which allows us to check that our code is correct and is supposed to help with the review. We shouldn't all write in typescript in ‘wow that's typescript’ mode but see it more as a tool to limit errors.

Again, they should not be in TypeScript. There are reasons here, and I'm not going to approve a PR just because you feel that something needs to be refactored to TypeScript.

**Review by ovflowd — CHANGES_REQUESTED** [review]
I appreciate your effort here, but sorry, I don't want this to get merged.

We shouldn't keep refactoring things on a whim, and there are reasons why certain things are not in TypeScript, they are internal utils and configuration files and hardly benefit anything from TypeScript.

**Comment by AugustinMauroy** [comment]
before closing, can we get opinion from other members of the team @nodejs/nodejs-website

File: apps/site/next-data/generators/releaseData.mjs
Comment (bmuenzenmeyer) [inline apps/site/next-data/generators/releaseData.mjs:4]:
```suggestion
// NOTE: this file cannot be a ts file
// because `nodevu` didn't have typedefs
```

File: apps/site/next-data/generators/websiteFeeds.ts
Comment (bmuenzenmeyer) [inline apps/site/next-data/generators/websiteFeeds.ts:37]: this is now out of date

File: apps/site/next-data/generators/websiteFeeds.ts
Comment (bmuenzenmeyer) [inline apps/site/next-data/generators/websiteFeeds.ts:43]:
```suggestion
      // @ts-expect-error - The addItem need a valid Item type
```

File: apps/site/next.dynamic.constants.ts
Comment (bmuenzenmeyer) [inline apps/site/next.dynamic.constants.ts:15]: this sorta implies their is a type we can import rather than define right?

File: apps/site/next.mdx.use.client.ts
Comment (bmuenzenmeyer) [inline apps/site/next.mdx.use.client.ts:36]:
```suggestion
  // @ts-expect-error - MDXImage is a wrapper of Next Image so it doesn't have same props as HTMLImageElement
```

**Review by bmuenzenmeyer — COMMENTED** [review]
There are certainly parts of this PR that need polish (typos), TODOs, but I guess I'd like to better understand why more TypeScript is bad. I see some typings in here improving the code quality, while others simply shorten imports or replace jsdoc. 

There is value here on both sides, but I don't yet understand Claudio's point of view.

**Comment by ovflowd** [comment]
I believe the main reasons I'm against TypeScript here are:

- I'm the sort of maintainer who prefers TypeScript on userland (Apps), whereas plain JavaScript within Libraries or Configuration files.
- Some of these files should be able to be used independently of TypeScript or anything behind Turbopack/Webpack; I don't see the need for TypeScript here. Otherwise, we need to introduce yet another build step.
  - It is the same reason why many files within the i18n package are plain JavaScript; we would introduce more dependencies.

This change is not a QoL update (IMO). Are these JSDocs fine, or is anyone having issues with them? I feel that having continuous refactoring is negative, and the time spent here on transforming these files into TypeScript could have been well spent elsewhere.

I don't necessarily have strong feelings here, but I won't remove my -1 until someone provides me a solid, convincing reason why TypeScript here would be better for these files, if not just a matter of preference.

**Comment by bmuenzenmeyer** [comment]
> the time spent here on transforming these files into TypeScript could have been well spent elsewhere.

This is perhaps the best argument - we have a lot on our plate with finite time. Everything we do is an investment, and it's hard to see the dividends or losses right away. I will stop with the analogy.

Considering the lack of consensus, I think the proper course of action is to close. @AugustinMauroy are you okay with that?

**Comment by AugustinMauroy** [comment]
I am ok with that 👌

**Pull request closed without merging**
